### ISSUE TITLE
Unable to resolve import "prop-types" (reboost)

### ISSUE THREAD

**Issue by jet10000** [sarsamurmu/reboost#50]
I want to test and install monaco-react, but I get the following errors

![image](https://user-images.githubusercontent.com/2258120/99864664-a9b75d80-2bdf-11eb-843d-78b4981e2756.png)

Monaco.jsx
```
import React from "react";
import ReactDOM from "react-dom";

import Editor from '@monaco-editor/react';

const App = () => <Editor height="90vh" language="javascript" />;

const monacoElement = document.getElementById("monaco");
ReactDOM.render(<App />, monacoElement);

```

**Comment by sarsamurmu** [comment]
If you look at the [package.json](https://github.com/suren-atoyan/monaco-react/blob/master/package.json#L34) of `@monaco-editor/react` you can see that `prop-types` is a `peerDependency`, so it means you have to manually install it. Please install it in your project using
```shell
npm i prop-types
# or
yarn add prop-types
```

**Comment by jet10000** [comment]
thank you.

**Comment by jet10000** [comment]
I don’t know what happened, I tried to solve it myself, but I don’t know how to deal with it.

![image](https://user-images.githubusercontent.com/2258120/99866777-e9d20c80-2bee-11eb-876b-69dbe260b8d8.png)

![image](https://user-images.githubusercontent.com/2258120/99866782-f35b7480-2bee-11eb-817c-a766202b52c8.png)

reboost.js

![image](https://user-images.githubusercontent.com/2258120/99866795-0ec67f80-2bef-11eb-93aa-f41dce0af2c5.png)

monaco.jsx

![image](https://user-images.githubusercontent.com/2258120/99866820-40d7e180-2bef-11eb-8d4a-8644aab4f4d5.png)

monaco.html

![image](https://user-images.githubusercontent.com/2258120/99866826-4cc3a380-2bef-11eb-8f75-39646d6ec197.png)

**Comment by jet10000** [comment]
I don’t know what happened, I tried to solve it myself, but I don’t know how to deal with it.

https://github.com/suren-atoyan/monaco-react/issues/139


![image](https://user-images.githubusercontent.com/2258120/99867417-3b30ca80-2bf4-11eb-9644-e4d7a69e8cf4.png)

![image](https://user-images.githubusercontent.com/2258120/99867430-58659900-2bf4-11eb-943a-d03d8afaf681.png)

**Comment by sarsamurmu** [comment]
In your template include these
```html
<script type="module" src="{{PATH_TO_SCRIPT}}"></script>
<script nomodule src="{{PATH_TO_SCRIPT}}"></script>
```
Both are required. Hope it fixes the issue.

**Comment by jet10000** [comment]
no effect

![image](https://user-images.githubusercontent.com/2258120/99868677-78e72080-2bff-11eb-8312-ac069df41b82.png)

**Comment by sarsamurmu** [comment]
This is caused by CommonJS modules. Can you try adding this to your options
```js
start({
  // ... other options

  commonJSInterop: {
    mode: 1
  }
})
```

**Comment by jet10000** [comment]
![image](https://user-images.githubusercontent.com/2258120/99869827-0ed37900-2c09-11eb-8602-4007682bc6e7.png)


![image](https://user-images.githubusercontent.com/2258120/99869821-04b17a80-2c09-11eb-867e-7937daaf6129.png)

![image](https://user-images.githubusercontent.com/2258120/99869841-214db280-2c09-11eb-867a-84ba43e32b49.png)

**Comment by sarsamurmu** [comment]
Actually, CommonJS interoperability with ES modules is a complex thing. I guess I have to rewrite the transformer code to support it. Until then all you can do is wait.

**Comment by sarsamurmu** [comment]
It looks like the problem comes from `@monaco-editor/react`. It created an import mess in its code which's why Reboost can't optimize it to work with ES Modules. I tried `react-monaco-editor` and it's working fine with some configuration.

Files
- `reboost.js`
  ```js
  const { start, builtInPlugins: { UsePlugin, FilePlugin } } = require('reboost');

  start({
    entries: [
      ['./src/index.jsx', './public/dist/index.js']
    ],
    commonJSInterop: {
      mode: 1
    },
    plugins: [
      // For loading fonts
      UsePlugin({
        include: /\.ttf/i,
        use: FilePlugin()
      })
    ],
  })
  ```
- `src/index.js`
  ```js
  import React from "react";
  import ReactDOM from "react-dom";

  import MonacoEditor from "react-monaco-editor";

  const App = () => <MonacoEditor height="90vh" language="javascript" />;

  const monacoElement = document.getElementById("monaco");
  ReactDOM.render(<App />, monacoElement);
  ```

So you can just use [`react-monaco-editor`](https://github.com/react-monaco-editor/react-monaco-editor) instead of `@monaco-editor/react`. There's no other way to fix this in Reboost.

**Comment by jet10000** [comment]
I adjusted it according to your code and tested it, but it still doesn't work. Can you add an example to the playground?

**Comment by jet10000** [comment]
![image](https://user-images.githubusercontent.com/2258120/99883322-b381a500-2c61-11eb-93f1-f2fed530a392.png)

**Comment by jet10000** [comment]
if enable commonJSInterop mode:1

![image](https://user-images.githubusercontent.com/2258120/99883393-2559ee80-2c62-11eb-86f6-e472951b22b0.png)

**Comment by sarsamurmu** [comment]
Oh, My bad. It needs a minor fix to the CSS plugin to work. So I can't do anything until I get my monitor back. Sorry.

**Comment by sarsamurmu** [comment]
Fixed in `v0.18.2`. Update your installation of Reboost.
